docs/reference/config.md

```diff
@@ -994,18 +994,6 @@ The following settings are available:
 `k8s.pullPolicy`
 : Defines the strategy to be used to pull the container image e.g. `'Always'`.
 
-`k8s.retryPolicy.delay`
-: Delay when retrying failed API requests (default: `500ms`).
-
-`k8s.retryPolicy.jitter`
-: Jitter value when retrying failed API requests (default: `0.25`).
-
-`k8s.retryPolicy.maxAttempts`
-: Max attempts when retrying failed API requests (default: `4`).
-
-`k8s.retryPolicy.maxDelay`
-: Max delay when retrying failed API requests (default: `90s`).
-
```

Comment:
it may be better to exclude K8s from this since its' a separate executor/plugin, also to avoid a breaking change in the current settings.  @bentsherman what do you think?

Reply:
Not sure ewe should go with this unification. The retry policy could be very different depending the infra for example azure requires much more retries compared to AWS. same could be for K8s depending how is deployed. Think it's preferable to keep those independent and have a common one for nextflow core runtime

Reply:
You're right, and it probably doesn't make sense for users to put retry settings in their pipeline config anyway.

I suggest we read these settings from the user config at `$NXF_HOME/nextflow.config` and document that certain settings like `nextflow.retryPolicy` are only read from this config.

I would also move the [scm config](https://nextflow.io/docs/latest/git.html#git-configuration) into the user config and make it a first-class config scope, as it is currently treated separately from other scopes and is not handled by the new config parser

Reply:
> I'm going to have a try to scm file approach

Using the scm file doesn't make sense because these settings are also used for file publishing. Better to make the scm config part of the user config

Reply:
i'm open to alternatives about retry settings but `scm` cannot be moved in the user config, both to avoid circular deps and usage made by platform